src/main/java/uk/gov/ons/ctp/response/collection/exercise/client/impl/ActionSvcRestClientImpl.java

```diff
@@ -160,6 +161,56 @@ public ActionPlanDTO getActionPlanBySelectors(
     return actionPlans.get(0);
   }
 
+  @Retryable(
+      value = {RestClientException.class},
+      maxAttemptsExpression = "#{${retries.maxAttempts}}",
+      backoff = @Backoff(delayExpression = "#{${retries.backoff}}"))
+  @Override
+  public List<ActionPlanDTO> getActionPlansBySelectorsSocial(final String collectionExerciseId) {
+    log.debug(
+        "Retrieving action plan for selectors, collectionExerciseId: {}", collectionExerciseId);
+
+    final MultiValueMap<String, String> queryParams = new LinkedMultiValueMap<>();
+    queryParams.add(SELECTOR_COLLECTION_EXERCISE_ID, collectionExerciseId);
+    final UriComponents uriComponents =
+        restUtility.createUriComponents(appConfig.getActionSvc().getActionPlansPath(), queryParams);
+
+    HttpEntity httpEntity = restUtility.createHttpEntity(null);
+
+    final ResponseEntity<List<ActionPlanDTO>> responseEntity;
+    responseEntity =
+        restTemplate.exchange(
+            uriComponents.toString(),
+            HttpMethod.GET,
+            httpEntity,
+            new ParameterizedTypeReference<List<ActionPlanDTO>>() {});
+
+    log.debug(
+        "Successfully retrieved action plan for selectors, collectionExerciseId: {}",
+        collectionExerciseId);
+    return responseEntity.getBody();
+  }
+
+  @Override
+  public ActionPlanDTO getActionPlanBySelectorsSocial(String collectionExerciseId)
+      throws CTPException {
+
+    final List<ActionPlanDTO> actionPlans = getActionPlansBySelectorsSocial(collectionExerciseId);
+
+    if (actionPlans == null) {
+      throw new CTPException(
+          Fault.RESOURCE_NOT_FOUND, String.format(FOUND_NO_ACTION_PLANS_2, collectionExerciseId));
+    }
+
+    if (actionPlans.size() != 1) {
```

Comment:
I think comments of specific expected number should be commented and documented for future reference